torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -52,21 +51,43 @@ TensorView::TensorView(const std::shared_ptr<c10::TensorType>& tensor_type)
     }
   }
 
+  // default to non_contiguous;
+  std::vector<bool> contig_info(tensor_type->dim().value(), false);
+
+  // we iterate through stride_index_, which goes from fastest changing
+  // dimension to slowest, instead of iterating through sizes. This allows
+  // easier contiguity check;
   for (decltype(tensor_type->dim().value()) i = 0;
        i < tensor_type->dim().value();
        i++) {
-    // TODO: this is a temporary WAR to avoid contiguous_ flag on broadcasted
-    // dim, which results in wrong indexing math.
-    if (sizes[i]->isBroadcast() ||
-        (i != tensor_type->dim().value() - 1 && sizes[i + 1]->isBroadcast())) {
-      contig_info.push_back(false);
-    } else {
-      if (tensor_type->stride_properties()[i].has_value() &&
-          tensor_type->stride_properties()[i]->contiguous_.has_value() &&
-          tensor_type->stride_properties()[i]->contiguous_.value() == true) {
-        contig_info.push_back(true);
+    // if we don't have contiguous dimension at current stride index, don't
+    // bother;
+    if (tensor_type->stride_properties()[i].has_value() &&
```

Comment:
`const auto& property = tensor_type->stride_properties()[i];` then use this variable instead of multiple lookups?